driver-core/src/main/java/com/datastax/driver/core/Metrics.java

```diff
@@ -273,6 +273,7 @@ void shutdown() {
         private final Counter writeTimeouts = registry.counter("write-timeouts");
```

Comment:
If you add methods to a class that someone added to their own sub-implementation, this could lead to unexpected behavior because the methods of the subclass now override the ones you added; but strictly speaking it's still binary compatible.
In that case I think it's reasonable to assume that nobody extended the class to add those methods.